src/libs/MoneyRequestUtils.ts

```diff
@@ -34,10 +34,10 @@ function addLeadingZero(amount: string): string {
 /**
  * Calculate the length of the amount with leading zeroes
  */
-function calculateAmountLength(amount: string): number {
+function calculateAmountLength(amount: string, decimals: number): number {
     const leadingZeroes = amount.match(/^0+/);
     const leadingZeroesLength = leadingZeroes?.[0]?.length ?? 0;
-    const absAmount = parseFloat((Number(stripCommaFromAmount(amount)) * 100).toFixed(2)).toString();
+    const absAmount = parseFloat((Number(stripCommaFromAmount(amount)) * 10 ** decimals).toFixed(2)).toString();
```

Comment:
This change caused issue https://github.com/Expensify/App/issues/42084, some context:
> In calculateAmountLength, we multiply the amount with 10 to the power of the number of decimals allowed on a currency. So, if the currency allows 2 decimals, then it will be multiplied by 100, that's why we can only input 8 digit length for currency like USD. But if the currency doesn't allow decimal, such as IQD, then we can input 10 digit length.
> 
> However, [in the BE, all amounts are in cent](https://github.com/Expensify/App/pull/5243/files#r708725517), so the 10-digit length will be 12 digit length, and BE will return an error for it.